XLS-0040d-decentralized-identity/README.md

```diff
@@ -0,0 +1,416 @@
+
+## 4.1. The `DID` Object
+We propose a new ledger object called **`DID`** that holds references to or data associated with a single DID. The `DID` object is created and updated using the `DIDSet` transaction and is deleted using the `DIDDelete` transaction. 
+
+### 4.1.1. Object Fields
+`DID` object may have the following required and optional fields. 
+
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `LedgerEntryType` | :heavy_check_mark: | `string`  |   `UINT16`  |
+
+Identifies the type of ledger object. The proposal recommends the value `0x0049` as the reserved entry type.
+
+---
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `PreviousTxnID` | :heavy_check_mark: | `string`  |   `Hash256`  |
+
+The identifying hash of the transaction that most recently modified this object.
+
+---
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `PreviousTxnLgrSeq` | :heavy_check_mark: | `number`  |   `UINT32`  |
+
+The index of the ledger that contains the transaction that most recently modified this object.
+
+---
+
+
+| Field Name        |     Required?      | JSON Type | Internal Type 
+| ----------------- | :----------------: | :-------: | :-----------: 
+| `URI`             |                    | `string`  |   `BLOB`    |
+
+`URI` specifies a Universal Resource Identifier, a.k.a. URI that SHOULD point to the corresponding DID document or to the data associated with the DID. This field could be an HTTP(S) URL or IPFS URI. The `URI` field is NOT checked for validity, but the field is limited to a maximum length of 256 bytes. 
+
+---
+| Field Name   |     Required?      | JSON Type | Internal Type 
+| -------------| :----------------: | :-------: | :-----------: 
+| `Data`       |                    | `string`  |   `BLOB`    |
+
+`Data` field SHOULD contain the public attestations of identity credentials associated with the DID. The `Data` field is NOT checked for validity, but the field is limited to a maximum length of 256 bytes. 
+
+---
+| Field Name   |     Required?      | JSON Type | Internal Type 
+| -------------| :----------------: | :-------: | :-----------: 
+| `DIDDocument`       |                    | `string`  |   `BLOB`    |
+
+`DIDDocument` field SHOULD contain the <a href="https://www.w3.org/TR/did-core/#did-documents"> DID document </a> per W3C standards associated with the DID. The field is NOT checked for validity, and is limited to a maximum length of 256 bytes.
+
+Note:
+- Either `URI`, `Data` or `DIDDocument` field MUST be present to create and update the `DID` object.
+- All `URI`, `Data` and `DIDDocument` fields MUST be absent to delete the `DID` object.
```

Comment:
Remind me why this must be true?

Reply:
This isn't true in the implementation - in fact, you can't have a DID object without any of those fields. This doesn't make sense to me either, but happy to change if there's a reason.

Reply:
This is for a transaction, not for an object. I can rephrase it if it's confusing.

Reply:
Maybe add another note that says something like "A valid DID object must have either a `URI` or `DIDDocument` field"?